client/sagas/placeSagas.js

```diff
@@ -1,23 +1,24 @@
-import { call, takeEvery, put } from 'redux-saga/effects';
+import { call, takeEvery, put, select } from 'redux-saga/effects';
 import { getPlaceIds } from 'services/placeApi';
 import { getRandom } from 'lib/utils';
-import placeActions from 'actions/placeActions';
-import {
-  FETCH_PLACES,
-} from 'actions/placeActionTypes';
+import PLACE_ACTIONS from 'actions/placeActionTypes';
+import { REQUEST } from 'actions/constants';
+import { fetchPlaceSuccess } from 'actions/placeActions';
+import { fetchPlaceError } from '../actions/placeActions';
 
-function* fetchPlace(action) {
+export function* fetchPlaces() {
   try {
-    const places = yield call(getPlaceIds, action.payload);
+    const condition = yield select(state => state.condition);
```

Comment:
for saga, we should avoid having the code dependent on store's state so its easier to modify the state without changing saga code

Reply:
I see my reason for the change is I didn't want the logic of connectkng the react component to redux merely to pass a part of redux state to the payload for the saga if it's already directly accessible in the saga. But I do understand the reasoning behind also if the saga is not dependent with the state, it becomes a more reusable saga I think. Is my understanding correct?